Backpack/src/main/java/net/skyscanner/backpack/flare/BpkFlare.kt

```diff
@@ -151,20 +175,31 @@ open class BpkFlare @JvmOverloads constructor(
       pointerPosition.offset
     }
 
-    val yStart = height - pointerMask.height
+    val (yEnd, yStart) = if (pointerDirection == PointerDirection.DOWN) {
+      Pair(height, height - pointerMask.height)
+    } else {
+      Pair(pointerMask.height.toFloat(), 0f)
+    }
+
     val pointerXStart = width * pointerOffset - pointerHalfWidth
     val pointerXEnd = width * pointerOffset + pointerHalfWidth
 
-    clipRect.set(0f, yStart, pointerXStart, height)
+    clipRect.set(0f, yStart, pointerXStart, yEnd)
     canvas.clipOutRectCompat(clipRect)
 
-    clipRect.set(width, yStart, pointerXEnd, height)
+    clipRect.set(width, yStart, pointerXEnd, yEnd)
     canvas.clipOutRectCompat(clipRect)
 
     super.draw(canvas)
     canvas.restoreToCount(count)
 
-    canvas.drawBitmap(pointerMask, pointerXStart, yStart, paint)
+    val mask = if (pointerDirection == PointerDirection.DOWN) {
```

Comment:
instead of allocating different masks you can just rotate here the existing one. E.g. something like
val count = canvas.save()
canvas.rotate(90)
canvas.drawBitmap()
canvas.restoreToCount(count)

Reply:
Done